src/components/Filters/FilterModal.tsx

```diff
@@ -1,243 +1,296 @@
+	editFilter?: JSONSchema;
+	addFilter: (filter: JSONSchema) => void;
+	onClose: () => void;
 }
 
-const defaultFilterCompareFn: FilterFieldCompareFn = (a, b) => {
-	return a.title.toLowerCase().localeCompare(b.title.toLowerCase());
-};
-
-const FilterInput = ({
-	schema,
-	value,
-	operator,
-	onUpdate,
-}: FilterInputProps) => {
-	const model = getDataModel(schema);
-	const uiSchema = model && 'uiSchema' in model ? model.uiSchema : {};
-	if (model && 'Edit' in model) {
-		const Edit = model.Edit as React.FC<any>;
-		return (
-			<Edit
-				schema={schema}
-				value={value}
-				operator={operator}
-				onUpdate={onUpdate}
-				slim
-			/>
-		);
-	}
-	return (
-		<Form
-			schema={schema}
-			value={value}
-			onFormChange={({ formData }) => onUpdate(formData)}
-			uiSchema={{ 'ui:options': { label: false, inline: true }, ...uiSchema }}
-			hideSubmitButton
-		/>
-	);
-};
-
-const RelativeBox = styled(Box)`
-	position: relative;
-`;
-
-const DeleteButton = styled(Button)`
-	color: rgba(0, 0, 0, 0.4);
-	position: absolute;
-	bottom: 7px;
-	right: -35px;
-`;
-
 export const FilterModal = ({
+	schema,
+	editFilter,
 	addFilter,
 	onClose,
-	schema,
-	edit,
-	fieldCompareFn,
 }: FilterModalProps) => {
-	const [filters, setFilters] = useState(edit);
-	const [searchTerm, setSearchTerm] = useState('');
-
-	const debouncedSetSearchTerm = debounce(setSearchTerm, 300);
-
-	const setEditField = (field: string, index: number) => {
-		const currentEdit = filters.map((filter, i) =>
-			i === index ? SchemaSieve.getCleanEditModel(schema, field) : filter,
+	const properties: PropertyOption[] | undefined = React.useMemo(() => {
+		if (!schema.properties || typeof schema.properties === 'boolean') {
+			return;
+		}
+		const constructedProperties: PropertyOption[] = [];
+		Object.entries(schema.properties).forEach(
+			([key, value]: [string, JSONSchema]) => {
+				const refScheme = getPropertyScheme(value);
+				if (!refScheme?.length) {
+					constructedProperties.push({
+						key,
+						field: key,
+						title: getSchemaTitle(value, key),
+					});
+				} else {
+					refScheme.forEach((ref: string) => {
+						constructedProperties.push({
+							key: `${key}__${ref}`,
+							field: key,
+							title: getSchemaTitle(value, key, ref),
+							refScheme: ref,
+						});
+					});
+				}
+			},
 		);
-		setFilters(currentEdit);
-		setSearchTerm('');
-	};
+		return constructedProperties.sort((a, b) => (a.title > b.title ? 1 : -1));
+	}, [schema]);
 
-	const setEditOperator = (operator: string, index: number) => {
-		const currentEdit = filters.map((filter, i) =>
-			i === index ? { ...filter, operator } : filter,
-		);
-		setFilters(currentEdit);
+	const newSignature = {
+		key: !!properties?.[0].refScheme
+			? `${properties?.[0].field}__${properties?.[0].refScheme}`
+			: properties?.[0].field,
+		title: properties?.[0].title,
+		field: properties?.[0].field,
+		operator: undefined,
+		value: undefined,
+		refScheme: properties?.[0].refScheme,
 	};
```

Comment:
Would it make more sense to just `return null;` or show a spinner in case `properties?.[0] == null`?
Is a `newSignature` with undefined props useful?